clients/src/test/java/org/apache/kafka/clients/consumer/internals/MembershipManagerImplTest.java

```diff
@@ -764,6 +766,44 @@ public void testLeaveGroupWhenMemberOwnsAssignment() {
         testLeaveGroupReleasesAssignmentAndResetsEpochToSendLeaveGroup(membershipManager);
     }
 
+    @Test
+    public void testLeaveGroupWhenAssignmentEmpty() {
+        String topicName = "topic1";
+        TopicPartition ownedPartition = new TopicPartition(topicName, 0);
+        MembershipManager membershipManager = createMemberInStableState();
+
+        // We own a partition and rebalance listener is registered
+        when(subscriptionState.assignedPartitions()).thenReturn(Collections.singleton(ownedPartition));
+        when(subscriptionState.hasAutoAssignedPartitions()).thenReturn(true);
+        when(subscriptionState.rebalanceListener()).thenReturn(Optional.of(mock(ConsumerRebalanceListener.class)));
+        doNothing().when(subscriptionState).markPendingRevocation(anySet());
+
+        // Trigger leave group
+        CompletableFuture<Void> leaveResult1 = membershipManager.leaveGroup();
+
+        // Rebalance listener not yet triggered
+        final ArgumentCaptor<ConsumerRebalanceListenerCallbackNeededEvent> consumerRebalanceListener =
+            ArgumentCaptor.forClass(ConsumerRebalanceListenerCallbackNeededEvent.class);
+        verify(backgroundEventHandler).add(consumerRebalanceListener.capture());
+        final ConsumerRebalanceListenerCallbackNeededEvent callbackEvent = consumerRebalanceListener.getValue();
+        assertFalse(leaveResult1.isDone());
+        assertEquals(MemberState.PREPARE_LEAVING, membershipManager.state());
+
+        // Clear the assignment
+        when(subscriptionState.assignedPartitions()).thenReturn(Collections.emptySet());
+        when(subscriptionState.hasAutoAssignedPartitions()).thenReturn(false);
+
+        // Complete the callback
+        callbackEvent.future().complete(null);
+
+        // Completed the listener
```

Comment:
This comment does not make too much sense since the leave future is still verified to be not completed.

Reply:
Done